packages/fetchai/connections/p2p_libp2p_client/connection.py

```diff
@@ -119,6 +97,14 @@ def __init__(self, **kwargs):
             "Delegate Uri should be provided for each node",
         )
 
+        nodes_certs = [node["cert"] for node in nodes]
+        enforce(
+            len(nodes_certs) == len(nodes),
+            "Delegate Cert should be provided for each node",
+        )
```

Comment:
If no `cert` is there it will throw a key error; should use `get` with default value instead, e.g:
```suggestion
        nodes_certs = [node["cert"] for node in nodes if node.get("cert") is not None]
        enforce(
            len(nodes_certs) == len(nodes),
            "Delegate Cert should be provided for each node",
        )
```